docs/register/registration.rst

```diff
@@ -140,6 +140,18 @@ The parameter ``c`` provided to the expression is the *component context* (an ``
 
 Additional dependencies can be satisfied using this context parameter - in the example, ``A`` requires a constructor parameter of type ``B`` that may have additional dependencies.
 
+As well as using ``IComponentContext`` to resolve dependencies in your lambda expression, you can also use the generic ``Register`` overloads to specify your dependencies as a variable number of typed arguments to the lambda, and Autofac will resolve them for you:
+
+.. sourcecode:: csharp
+
+  builder.Register((IDependency1 dep1, IDependency2 dep2) => new Component(dep1, dep2));
+
+You can blend the ``IComponentContext`` and generic approach if you need to make conditional choices, or use methods like ``ResolveNamed``:
```

Comment:
```suggestion
You can blend the ``IComponentContext`` and generic approach if you need to make conditional choices, or use methods like ``ResolveNamed``. Just add the ``IComponentContext`` as the first parameter to the lambda:
```

Figured it might be worth clarifying that folks probably shouldn't just jam it at the end or whatever. With 20 overloads (!) now for the function, it might not be obvious.